zlib.c

```diff
@@ -38,30 +38,20 @@ static void zlib_pre_call(git_zstream *s)
 {
 	s->z.next_in = s->next_in;
 	s->z.next_out = s->next_out;
-	s->z.total_in = s->total_in;
-	s->z.total_out = s->total_out;
+	s->z.total_in = 0;
+	s->z.total_out = 0;
 	s->z.avail_in = zlib_buf_cap(s->avail_in);
 	s->z.avail_out = zlib_buf_cap(s->avail_out);
 }
 
 static void zlib_post_call(git_zstream *s)
 {
-	size_t bytes_consumed;
-	size_t bytes_produced;
-
-	bytes_consumed = s->z.next_in - s->next_in;
-	bytes_produced = s->z.next_out - s->next_out;
-	if (s->z.total_out != s->total_out + bytes_produced)
-		BUG("total_out mismatch");
-	if (s->z.total_in != s->total_in + bytes_consumed)
-		BUG("total_in mismatch");
-
-	s->total_out = s->z.total_out;
-	s->total_in = s->z.total_in;
+	s->total_out += (size_t) s->z.total_out;
```

Comment:
implementation defined. we should be explicit.

Reply:
No, I disagree. If you are really serious about being explicit, you even need to say

```c
for (i = (int)0; i < (int)10; i++)
```

which is pretty ridiculous an idea.

In this particular instance, the extra `(size_t)` cast does *nothing* else than the same code *without* that cast.

Therefore it is useless churn.

It is actually worse than that, even: not only does it drown out the real changes in the diff (which makes reviews hard, and bug easy to slip in, both things that we do want to avoid, right?), it also makes for a real head-scratcher when a seasoned C developer comes along and stumbles over the cast ("Why is that there?"). So it is a triple whammy of badness.

Let's not do that. Let's keep with Git's coding conventions, drop the unnecessary cast, and be done with it.

I mean, we could continue this conversations for days, weeks, even months, but the essential value of it already dropped pretty much to zero, at least from my point of view.